wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -1271,6 +1301,71 @@ class ValidatedBlock(blocks.StreamBlock):
             3: ['Enter a valid URL.'],
         })
 
+    def test_block_level_validation_no_errors(self):
```

Comment:
You're right, this is needlessly complicated. I was attempting to test that the validation logic is called, and any ValidationErrors raised behave as expected. I added tests for a use case that closely matches a real use case that I wrote this PR for, but I can make something simpler. I can also write a test to ensure the error messages actually appear in the rendered form.